### PR TITLE
Migration stuff, examples and some fixes for found issues

### PR DESCRIPTION
**Pull request by rase-**
- Migration script
- Examples
- Only index basic fields if the value changed
- Parents/children auto cleanup fixes
- More flexible URL regex

### CODE REVIEW

File: docs/query.md [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+  - [**$filter**]($filter)
+  - [**$or**]($or)
+  - [**$and**]($and)
+
+
+[Available data types](#available-data-types)
+
+## Clauses
+
+### Default fields
```
Comment (rase-) [inline docs/query.md]: There's a few more default fields:

```
export const defaultFields: Record<string, FieldSchema> = {
  id: {
    type: 'id'
    // never indexes these - uses in keys
    // search: { index: 'default', type: ['TAG'] }
  },
  type: {
    search: { index: 'default', type: ['TAG'] },
    type: 'type'
  },
  name: {
    search: { index: 'default', type: ['TAG'] },
    type: 'string'
  },
  children: {
    type: 'references'
  },
  parents: {
    type: 'references'
  },
  ancestors: {
    type: 'references',
    search: { index: 'default', type: ['TAG'] }
  },
  descendants: {
    type: 'references'
  },
  aliases: {
    type: 'set',
    items: { type: 'string' }
  }
}

```

File: docs/query.md [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+const result = await get({
+  $id: 'moASxsd3',
+  director: { $default: 'Unknown director' }
+})
+```
+[See test](../client/test/examples/clauses/default.ts)
+
+### `$language`: _string_
+
+Filters document data to a set language on fields that supoprt it.
```
Comment (rase-) [inline docs/query.md]: spoport -> support

File: docs/schemas.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  - `EXISTS`
+  - `TAG`
+  - `TEXT`
+  - `NUMERIC`
+  - `SORTABLE`
+  - `TEXT-LANGUAGE`
+  - `GEO`
+  - `TEXT-LANGUAGE-SUG`
+
+#### Field data types
```
Comment (rase-) [inline docs/schemas.md]: We also support set/array/json field types

File: docs/query.md [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+```javascript
+const = await client.get({
+  $id: 'mo2001ASpaceOdyssey',
+  by: { $field: ['producer', 'director'] }
+})
+```
+
+[See test](../client/test/examples/clauses/field.ts)
+
+### `$inherit`: _boolean_
```
Comment (rase-) [inline docs/query.md]: Here we could also explain by type/name inheritance and inherit as item

Comment (nfrade) [reply docs/query.md]: how does that work? have an example?

Comment (rase-) [reply docs/query.md]: Inherit item: https://github.com/atelier-saulx/selva/blob/master/client/test/getBasic.ts#L1021

Comment (rase-) [reply docs/query.md]: By type: https://github.com/atelier-saulx/selva/blob/master/client/test/getField.ts#L479

Comment (rase-) [reply docs/query.md]: By name: https://github.com/atelier-saulx/selva/blob/master/client/test/getField.ts#L530 (since name is also non-unique)

**Pull request merged**